great_expectations/experimental/datasources/interfaces.py

```diff
@@ -15,29 +15,38 @@
     Type,
     TypeVar,
     Union,
+    overload,
 )
 
+import pandas as pd
 import pydantic
-from pydantic import Field
+from pydantic import Field, StrictBool, StrictInt
 from pydantic import dataclasses as pydantic_dc
-from typing_extensions import ClassVar, TypeAlias, TypeGuard
+from pydantic import root_validator, validate_arguments
+from typing_extensions import ClassVar, Literal, TypeAlias, TypeGuard
 
+from great_expectations.core.id_dict import BatchSpec
 from great_expectations.experimental.datasources.experimental_base_model import (
     ExperimentalBaseModel,
 )
 from great_expectations.experimental.datasources.metadatasource import MetaDatasource
 from great_expectations.experimental.datasources.sources import _SourceFactories
+from great_expectations.validator.metric_configuration import MetricConfiguration
 
 LOGGER = logging.getLogger(__name__)
 
 if TYPE_CHECKING:
-    from great_expectations.core.batch import (
-        BatchDataType,
-        BatchDefinition,
-        BatchMarkers,
-        BatchSpec,
-    )
+    # TODO: When the Batch class is moved out of the experimental directory, we should try to import the annotations
+    #       from core.batch so we no longer need to call Batch.update_forward_refs() before instantiation.
+    from great_expectations.core.batch import BatchData, BatchDefinition, BatchMarkers
     from great_expectations.execution_engine import ExecutionEngine
+    from great_expectations.validator.computed_metric import MetricValue
+
+try:
```

Comment:
Can you move this `try` into the `TYPE_CHECKING` block? 

_The rest of the comment is about this pattern of setting the import to None and isn't relevant when we only use the imported values for typing._

I know we use this pattern a bit but I think we should get away from it when we are using values from these maybe imported modules. That is we should define a `MaybeImported` type that wraps this import so we can give better error messages when something like this happens: https://greatexpectationslabs.slack.com/archives/C026YSF2JA3/p1673453243284299